pkg/asset/installconfig/powervs/metadata.go

```diff
@@ -37,34 +37,49 @@ type Metadata struct {
 	dnsInstanceCRN string
 	client         *Client
 
-	mutex sync.Mutex
+	mutex       sync.Mutex
+	clientMutex sync.Mutex
 }
 
 // NewMetadata initializes a new Metadata object.
 func NewMetadata(config *types.InstallConfig) *Metadata {
 	return &Metadata{BaseDomain: config.BaseDomain, PublishStrategy: config.Publish}
 }
 
+// Client returns a client used for making API calls to IBM PowerVS services.
+func (m *Metadata) Client() (*Client, error) {
```

Comment:
I don't understand why do you have to store the client and expose it via the Metadata type if the client can be easily created without any arguments with `NewClient`?

If you were to expose it via `Metadata`, why can't you just initialize it once in `NewMetadata`? That way you don't need a mutex for the client.